geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HScanDunitTest.java

```diff
@@ -0,0 +1,178 @@
+
+  private static Properties locatorProperties;
+
+  private static MemberVM locator;
+  private static MemberVM server1;
+  private static MemberVM server2;
+  private static MemberVM server3;
+
+  private static int redisServerPort1;
+  private static int redisServerPort2;
+  private static int redisServerPort3;
+
+  private final int SIZE_OF_INITIAL_HASH_DATA = 1000;
+  private final Map<String, String> INITIAL_HASH_DATA = makeEntrySet(SIZE_OF_INITIAL_HASH_DATA);
+  final String HASH_KEY = "key";
+  final String BASE_FIELD = "baseField_";
+
+  @BeforeClass
+  public static void classSetup() {
+    locatorProperties = new Properties();
+    locatorProperties.setProperty(MAX_WAIT_TIME_RECONNECT, "15000");
+    locator = clusterStartUp.startLocatorVM(0, locatorProperties);
+    int locatorPort = locator.getPort();
+
+    server1 = clusterStartUp
+        .startRedisVM(1,
+            x -> x.withProperty(ConfigurationProperties.SERIALIZABLE_OBJECT_FILTER,
+                "org.apache.commons.lang3.tuple.**;org.apache.geode.**")
+                .withConnectionToLocator(locatorPort));
+
+    server2 = clusterStartUp
+        .startRedisVM(2,
+            x -> x.withProperty(ConfigurationProperties.SERIALIZABLE_OBJECT_FILTER,
+                "org.apache.commons.lang3.tuple.**;org.apache.geode.**")
+                .withConnectionToLocator(locatorPort));
+
+    server3 = clusterStartUp
+        .startRedisVM(3,
+            x -> x.withProperty(ConfigurationProperties.SERIALIZABLE_OBJECT_FILTER,
+                "org.apache.commons.lang3.tuple.**;org.apache.geode.**")
+                .withConnectionToLocator(locatorPort));
+
+    redisServerPort1 = clusterStartUp.getRedisPort(1);
+    redisServerPort2 = clusterStartUp.getRedisPort(2);
+    redisServerPort3 = clusterStartUp.getRedisPort(3);
+
+    jedis1 = new Jedis(LOCAL_HOST, redisServerPort1, JEDIS_TIMEOUT);
+    jedis2 = new Jedis(LOCAL_HOST, redisServerPort2, JEDIS_TIMEOUT);
+    jedis3 = new Jedis(LOCAL_HOST, redisServerPort3, JEDIS_TIMEOUT);
+  }
+
+  @Before
+  public void testSetup() {
+    jedis1.flushAll();
+    jedis2.flushAll();
+    jedis3.flushAll();
+
+    jedis1.hset(HASH_KEY, INITIAL_HASH_DATA);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    jedis1.disconnect();
+    jedis2.disconnect();
+    jedis3.disconnect();
+
+    server1.stop();
+    server2.stop();
+    server3.stop();
+  }
+
+  @Test
+  public void should_notLoseFields_givenConcurrentThreadsDoingHScansAndChangingValues() {
+    final int ITERATION_COUNT = 1000;
+
+    List<Map.Entry<String, String>> allEntries1 = new ArrayList<>();
+    List<Map.Entry<String, String>> allEntries2 = new ArrayList<>();
+
+    new ConcurrentLoopingThreads(ITERATION_COUNT,
+        (i) -> multipleHScanLamba(allEntries1, jedis1),
+        (i) -> multipleHScanLamba(allEntries2, jedis2),
+        (i) -> jedis3.hset(HASH_KEY, BASE_FIELD + i, "new_value_" + i)).run();
+
+    assertThat(allEntries1.size())
```

Comment:
I'm not sure this is quite what you want. Since `multipleHScanLamba` always clears the entries, this assert is only ever checking the result of the last iteration and not all of them. It might be better it simply count the results in the lambda and then assert on `ITERATION_COUNT * SIZE_OF_INITIAL_HASH_DATA`